common-types/src/main/java/org/apache/falcon/entity/v0/Entity.java

```diff
@@ -82,7 +84,11 @@ public String toString() {
     public static Entity fromString(EntityType type, String str) {
         try {
             Unmarshaller unmarshaler = type.getUnmarshaller();
-            return (Entity) unmarshaler.unmarshal(new StringReader(str));
+            XMLInputFactory xif = XMLInputFactory.newFactory();
+            xif.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);
```

Comment:
Can you extract this editing before giving back to user to a method and so can be reused all places.